components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/siddhi/SiddhiProvider.java

```diff
@@ -0,0 +1,174 @@
+        siddhiDataProviderConfig.setQuery(((JsonObject) jsonElement).get(STORE_QUERY).getAsString());
+        siddhiDataProviderConfig.setSiddhiAppContext(((JsonObject) jsonElement).get(SIDDHI_APP).getAsString());
+        super.init(topic, sessionId, siddhiDataProviderConfig);
+        SiddhiAppRuntime siddhiAppRuntime = getSiddhiAppRuntime(topic);
+        siddhiAppRuntime.start();
+        StoreQuery storeQuery = SiddhiCompiler.parseStoreQuery(siddhiDataProviderConfig.getQuery());
+        Attribute[] outputAttributeList = siddhiAppRuntime.getStoreQueryOutputAttributes(storeQuery);
+        metadata = new DataSetMetadata(outputAttributeList.length);
+        Attribute outputAttribute;
+        for (int i = 0; i < outputAttributeList.length; i++) {
+            outputAttribute = outputAttributeList[i];
+            metadata.put(i, outputAttribute.getName(), getMetadataTypes(outputAttribute.getType().toString()));
+        }
+        return this;
+    }
+
+    @Override
+    public boolean configValidator(ProviderConfig providerConfig) throws DataProviderException {
+        SiddhiDataProviderConfig siddhiDataProviderConfig = (SiddhiDataProviderConfig) providerConfig;
+        return siddhiDataProviderConfig.getSiddhiAppContext() != null && siddhiDataProviderConfig.getQuery() != null;
+    }
+
+    @Override
+    public String providerName() {
+        return PROVIDER_NAME;
+    }
+
+    @Override
+    public DataSetMetadata dataSetMetadata() {
+        return metadata;
+    }
+
+    @Override
+    public String providerConfig() {
+        Map<String, String> renderingTypes = new HashMap();
+        renderingTypes.put(SIDDHI_APP, InputFieldTypes.SIDDHI_CODE);
+        renderingTypes.put(STORE_QUERY, InputFieldTypes.SIDDHI_CODE);
+        renderingTypes.put(PURGING_INTERVAL, InputFieldTypes.NUMBER);
+        return new Gson().toJson(new Object[]{renderingTypes, new SiddhiDataProviderConfig()});
+    }
+
+    @Override
+    public void publish(String topic, String sessionId) {
+        Event[] events = siddhiAppRuntime.query(siddhiDataProviderConfig.getQuery());
+        ArrayList<Object[]> data = new ArrayList<>();
+        for (Event event : events) {
+            data.add(event.getData());
+        }
+        publishToEndPoint(data, sessionId, topic);
+    }
+
+    @Override
+    public void purging() {
+        //In siddhi-store provider, we do not have a requirement to purge the data.
+    }
+
+    @Override
+    public void setProviderConfig(ProviderConfig providerConfig) {
+        this.siddhiDataProviderConfig = (SiddhiDataProviderConfig) providerConfig;
+
+    }
+
+    @Override
+    public DataSetMetadata getMetadata() {
+        return metadata;
+    }
+
+    @Override
+    public void stop() {
+        siddhiAppRuntime.shutdown();
+        super.stop();
+    }
+
+    private static SiddhiManager getSiddhiManager() {
+        if (siddhiManager == null) {
```

Comment:
This logic always returns`new SiddhiManager` instead of the static object.

Reply:
Fixed